routes/api/v1/forecasts.js

```diff
@@ -11,50 +11,27 @@ const fetch = require('node-fetch')
 const ForecastService = require('../../../app/services/forecast_service')
 
 router.get('/', (req, res) => {
-  // let users = database('users').select().then((users) => { console.log(users)})
   async function getUser() {
     let user = await database('users').where({api_key: req.body.api_key}).first()
     return user
   }
   getUser()
     .then((user) => {
-      if (user.api_key == req.body.api_key) {
-        // error out for no location - error code 422?
+      if (user.api_key == req.body.api_key && req.query.location) {
         async function getForecast() {
           let forecast = await ForecastService.getFullForecast(req.query.location);
           res.status(200).send(forecast);
         }
         getForecast()
-      } else {
+      } else if (!req.body.api_key ){
         res.status(401).json({error_message: 'Unauthorized request' })
+      } else if (!req.query.location){
+        res.status(400).json({error_message: 'Location required' })
       }
     })
     .catch((error) => {
       console.log({error:error, location: 'forecast get'});
```

Comment:
Maybe change the `location: 'location get'` part of the error. I don't understand it on a quick glance but it probably isn't a big deal.

Reply:
@BabsLabs 
It was to help see where my errors were coming from, just a label in the console essentially, but I will be taking that out!